lib/internal/process/main_thread_only.js

```diff
@@ -154,6 +154,13 @@ function setupSignalHandlers(internalBinding) {
       delete signalWraps[type];
     }
   });
+
+  // re-arm pre-existing signal event registrations
+  // with this signal wrap capabilities.
+  process.eventNames().forEach((ev) => {
```

Comment:
I thought we prefer for-of loops over `forEach`? (just a question)

Reply:
Just an observation, at least I thought we prefer for-of loops...
(on a side note, I took a look at the bytecode generated with for-of loops and they look humongous compared to forEach loops)

Reply:
from perf standpoint, forEach is usually faster in newer versions of V8. from from a semantic standpoint i think it looks better to use forEach, but that's just opinion.

Reply:
thanks @joyeecheung @devsnek for the reasoning; so I am keeping things as is.